ExtractorUtils/Unstable/Configuration/Connection.cs

```diff
@@ -31,24 +32,14 @@ public class ConnectionConfig : VersionedConfig
         /// <summary>
         /// ID of integration in CDF, required.
         /// </summary>
-        public string? Integration { get; set; }
+        public IntegrationConfig? Integration { get; set; }
 
         /// <summary>
-        /// Configuration for retries of failed requests to CDF.
+        /// Configuration for the connection to CDF.
         /// </summary>
-        public RetryConfig CdfRetries { get => _cdfRetries; set { _cdfRetries = value ?? _cdfRetries; } }
-        private RetryConfig _cdfRetries = new RetryConfig();
+        public CdfConnectionConfig CdfConnection { get => _cdfConnection; set { _cdfConnection = value ?? _cdfConnection; } }
```

Comment:
why `_cdfConnection = value ?? _cdfConnection;` here?
it won't allow resetting the value. is it intentional?

Reply:
It won't allow setting the value to null, which is intentional. It's the only real way to have a non-nullable property with yaml, that I have found.